picard/ui/options/interface.py

```diff
@@ -113,4 +157,97 @@ def starting_directory_browse(self):
+        list_item.setToolTip(_(u'Drag and Drop to re-order'))
+        if action != self.SEPERATOR:
+            list_item.setIcon(icontheme.lookup(self._get_icon_from_name(action), icontheme.ICON_SIZE_MENU))
+        if index:
+            self.ui.toolbar_layout_list.insertItem(index, list_item)
+        else:
+            self.ui.toolbar_layout_list.addItem(list_item)
+        return list_item
+
+    def _all_list_items(self):
+        return [self.ui.toolbar_layout_list.item(i).text() for i in range(self.ui.toolbar_layout_list.count())]
+
+    def _added_actions(self):
+        actions = self._all_list_items()
+        actions = filter(lambda x: x != self.SEPERATOR, actions)
+        return set(actions)
+
+    def _current_item(self, return_type='item'):
+        if return_type == 'item':
+            return self.ui.toolbar_layout_list.currentItem()
+        elif return_type == 'row':
+            return self.ui.toolbar_layout_list.currentRow()
+
+    def populate_action_list(self):
+        self.ui.toolbar_layout_list.setDragDropMode(QtGui.QAbstractItemView.DragDrop)
+        self.ui.toolbar_layout_list.setDefaultDropAction(QtCore.Qt.MoveAction)
+        self.ui.toolbar_layout_list.clear()
+        for name in config.setting['toolbar_layout']:
+            if name in self.TOOLBAR_BUTTONS.keys():
+                self._insert_item(self.TOOLBAR_BUTTONS[name])
+            else:
+                self._insert_item(self.SEPERATOR)
+
+    def update_buttons(self):
+        self.ui.add_button.setEnabled(self._added_actions() != self.ACTION_NAMES)
+        current_row = self._current_item('row')
+        self.ui.up_button.setEnabled(current_row > 0)
+        self.ui.down_button.setEnabled(current_row < self.ui.toolbar_layout_list.count() - 1)
+
+    def add_to_toolbar(self):
+        added_items = self._added_actions()
+        display_list = list(set.difference(self.ACTION_NAMES, added_items))
+        action, ok = QtGui.QInputDialog.getItem(self, "Add Action", "Select an Action:", display_list)
+        if ok:
+            list_item = self._insert_item(action)
+            self.ui.toolbar_layout_list.setCurrentItem(list_item)
+        self.update_buttons()
+
+    def insert_seperator(self):
+        insert_index = self._current_item('row') + 1
+        self._insert_item(self.SEPERATOR, insert_index)
+
+    def move_item(self, offset):
+        current_index = self._current_item('row')
+        offset_index = current_index - offset
+        offset_item = self.ui.toolbar_layout_list.item(offset_index)
+        if offset_item:
+            current_item = self.ui.toolbar_layout_list.takeItem(current_index)
+            self.ui.toolbar_layout_list.insertItem(offset_index, current_item)
+            self.ui.toolbar_layout_list.setCurrentItem(current_item)
+            self.update_buttons()
+
+    def remove_action(self):
+        item = self.ui.toolbar_layout_list.takeItem(self._current_item('row'))
+        del item
```

Comment:
actually python throws an error saying it cannot delete a function call other wise